pkg/security/probe/process_killer.go

```diff
@@ -162,8 +203,28 @@ func (p *ProcessKiller) isRuleAllowed(rule *rules.Rule) bool {
 	return slices.Contains(p.sourceAllowed, rule.Policy.Source)
 }
 
+// called once a rule got disarmed
+func (p *ProcessKiller) updateKillQueueAlarm(disarmer *ruleDisarmer) {
+	// check if we have another rule with queued kills and reset the alarm to
+	// the correct value. Disable the alarm otherwise.
+	// NB: we should not kill anything, but if we do it's ok
+	if alarm := p.KillQueuedPidsAndGetNextAlarm(); alarm != nil {
```

Comment:
should we split this function into 2 functions : one for killing another one for getting the nextAlarm. It is a bit confusing to see a function that aims to kill here (even there a check within the function itself).

Reply:
The idea it's to iterate only once and prevent races when setting a new alarm. I'd rather keep it as it